src/components/TodoForm/TodoForm.tsx

```diff
@@ -0,0 +1,107 @@
+import { useState } from 'react';
+
+import { getAllUsers, getUserById } from '../../services/user';
+import { isValidLetter } from '../../services/Pattern';
+import { Todo } from '../../types/Todo';
+
+interface Props {
+  onAdd: (todo: Todo) => void;
+  newTodoId: number;
+}
+
+export const TodoForm: React.FC<Props> = ({ onAdd, newTodoId }) => {
+  const [count, setCount] = useState(0);
+
+  const [title, setTitle] = useState('');
+  const [userId, setUserId] = useState(0);
+
+  const [hasTitleError, setHasTitleError] = useState(false);
+  const [hasUserIdError, setHasUserIdError] = useState(false);
+
+  const handleTitleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const TitleTetters = event.target.value;
+
+    if (isValidLetter(TitleTetters.slice(-1))) {
+      setTitle(TitleTetters);
+    } else {
+      setTitle(TitleTetters.slice(0, -1));
+    }
+
+    setHasTitleError(false);
+  };
+
+  const handleUserIdChange = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    setUserId(+event.target.value);
+    setHasUserIdError(false);
+  };
+
+  const reset = () => {
+    setTitle('');
+    setUserId(0);
+
+    setHasTitleError(false);
+    setHasUserIdError(false);
+
+    setCount(count + 1);
+  };
+
+  const handleAdd = (event: React.FormEvent) => {
+    event.preventDefault();
+
+    setHasTitleError(!title);
```

Comment:
you need to handle scenario when title contains spaces only. I shouldn't be able to add todo in this case

Reply:
<img width="209" alt="image" src="https://github.com/mate-academy/react_add-todo-form/assets/70220667/2289eae1-668a-4a01-aa08-172a15682329">
     
<img width="588" alt="image" src="https://github.com/mate-academy/react_add-todo-form/assets/70220667/a4ca671c-3d67-4fc1-869d-11a1baf91aa2">
still not fixed